util/syn_yosys.sh

```diff
@@ -10,59 +10,139 @@
+sed -i 's/\[(dm::Data0):DataEnd\]:/dm::Data0:/g' dm_csrs.sv
+sed -i 's/\[(dm::ProgBuf0):ProgBufEnd\]:/dm::ProgBuf0:/g' dm_csrs.sv
+sed -i 's/inside//g' dm_mem.sv
+sed -i 's/\[(dm::DataAddr):DataEndAddr\]:/dm::DataAddr:/' dm_mem.sv
+sed -i 's/\[DataBaseAddr:DataEndAddr\]:/DataBaseAddr:/' dm_mem.sv
+sed -i 's/\[ProgBufBaseAddr:ProgBufEndAddr\]:/ProgBufBaseAddr:/' dm_mem.sv
+sed -i 's/\[AbstractCmdBaseAddr:AbstractCmdEndAddr\]:/AbstractCmdBaseAddr:/' dm_mem.sv
+sed -i 's/\[FlagsBaseAddr:FlagsEndAddr\]:/FlagsBaseAddr:/' dm_mem.sv
+
+printf "\nSV2V VERSION:\n"
+sv2v --version
+
+printf "\nSV2V ERRORS:\n"
 for file in *.sv; do
   module=`basename -s .sv $file`
-  echo $file
-  sv2v --define=VERILATOR --define=SYNTHESIS *_pkg.sv prim_assert.sv $file > ${module}.v
-
-  # TODO: eventually remove below hack. It removes "unsigned" from params
-  # because Yosys doesn't support unsigned parameters
-  sed -i 's/parameter unsigned/parameter/g' ${module}.v
-  sed -i 's/localparam unsigned/localparam/g' ${module}.v
+  sv2v --define=SYNTHESIS *_pkg.sv prim_assert.sv $file > ${module}.v
 done
 
-# remove *pkg.v files (they are empty files and not needed)
+# postprocess generated Verilog to fix some more sv2v issues:
+# TODO: remove below hacks eventually
+sed -i "s/1'sb0/'d0/g" *.v
+sed -i "s/1'sb1/-'sd1/g" *.v
+sed -i 's/parameter ResetValue/parameter [0:0] ResetValue/' prim_flop_2sync.v
+
+# remove *pkg.v files (they are empty files and not needed for Verilog)
 \rm -Rf *_pkg.v
 
 #-------------------------------------------------------------------------
 # run LEC (generarted Verilog vs. original SystemVerilog)
 #-------------------------------------------------------------------------
 printf "\n\nLEC RESULTS:\n"
-cd ../../hw/formal
-for file in ../../util/syn_out/*.v; do
-  module=`basename -s .v $file`
-  lec_sv2v ../../util/syn_out $module > /dev/null 2>&1
+
+# top_earlgrey and all its submodules
+declare -a modules=(
+  "rv_dm"
+  "spi_device"
+  "usbdev"
+  "flash_ctrl"
+  "tlul_adapter_sram"
+  "prim_rom_adv"
+  "prim_ram_1p_adv"
+  "uart"
+  "gpio"
+  "aes"
+  "hmac"
+  "pinmux"
+  "padctrl"
+  "alert_handler"
+  "pwrmgr"
+  "rstmgr"
+  "clkmgr"
+  "nmi_gen"
+  "rv_timer"
+  "rv_plic"
+  "rv_core_ibex"
+  "xbar_main"
+  "xbar_peri"
+  "flash_phy"
+  "top_earlgrey"
+)
+
+# uncomment below two lines if you want to run LEC on all modules
+# for file in *.v; do
+#   export LEC_TOP=`basename -s .v $file`
+
+for module in "${modules[@]}"; do
+  export LEC_TOP="$module"
+
+  # run Conformal LEC
+  lec -xl -nogui -nobanner \
+    -dofile  ../../hw/formal/lec_sv2v.do \
+    -logfile lec_${LEC_TOP}.log \
+    <<< "exit -force" > /dev/null 2>&1
 
   # summarize results
-  result=`grep "Compare Results" lec_${module}.log`
+  result=`grep "Compare Results" lec_${LEC_TOP}.log`
   if [ $? -ne 0 ]; then
     result="CRASH"
   else
     result=`echo $result | awk '{ print $4 }'`
   fi
-  printf "%-25s %s\n" $module $result
+  printf "%-25s %s\n" $LEC_TOP $result
```

Comment:
I would suggest using `env` to explicitly set `LEC_TOP` for just the `lec` command (which I assume needs it?). Then leave the other uses of `$module` unchanged. Otherwise, you end up with two different variable names doing the same thing, which is always a little confusing.

Reply:
Good suggestion, will consider it for the new PR